cmake/BLTPrivateMacros.cmake

```diff
@@ -77,6 +77,57 @@ macro(blt_find_executable)
 endmacro(blt_find_executable)
 
 
+##------------------------------------------------------------------------------
+## blt_inherit_target_info( TO       <target>
+##                          FROM     <target>)
+##
+##  The purpose of this macro is if you want to grab all the inheritable info
+##  from the FROM target but don't want to make the TO target depend on it.
+##  Which is useful if you don't want to export the FROM target.
+##
+##------------------------------------------------------------------------------
+macro(blt_inherit_target_info)
+    set(options)
+    set(singleValueArgs TO FROM)
+    set(multiValueArgs)
+
+    # Parse the arguments
+    cmake_parse_arguments(arg "${options}" "${singleValueArgs}"
+                        "${multiValueArgs}" ${ARGN} )
+
+    # Check arguments
+    if ( NOT DEFINED arg_TO )
+        message( FATAL_ERROR "Must provide a TO argument to the 'blt_inherit_target' macro" )
+    endif()
+
+    if ( NOT DEFINED arg_FROM )
+        message( FATAL_ERROR "Must provide a FROM argument to the 'blt_inherit_target' macro" )
+    endif()
+
+    get_target_property(_interface_system_includes
+                        ${arg_FROM} INTERFACE_SYSTEM_INCLUDE_DIRECTORIES)
+    if ( _interface_system_includes )
+        blt_add_target_includes(TO       ${arg_TO}
+                                INCLUDES ${_interface_system_includes}
+                                SYSTEM   TRUE)
+    endif()
+
+    get_target_property(_interface_includes
+                        ${arg_FROM} INTERFACE_INCLUDE_DIRECTORIES)
+    if ( _interface_includes )
+        blt_add_target_includes(TO       ${arg_TO}
+                                INCLUDES ${_interface_includes})
+    endif()
+
+    get_target_property(_interface_defines
+                        ${arg_FROM} INTERFACE_COMPILE_DEFINITIONS)
+    if ( _interface_defines )
+        target_compile_definitions( ${arg_TO} PUBLIC ${_interface_defines})
+    endif()
+
```

Comment:
Would we ever need link flags or link directories (or anything else)?

Reply:
First,  this is an awesome new macro. 

Second, i do think we may want linking info in the future - don't think we need it yet for this PR, so we can we create a ticket as a reminder?